### PR TITLE
feat: support `no_std` for `reth-primitives`

### LINKED ISSUE #5746: Split `reth-primitives` and support `no_std` compilation.
### Describe the feature

I noticed that reth-primitives can be used by the rust eth ecosystem and even supports no_std compilation. Is there any plan to support splitting this crate into several crates? I could help to improve this.

### Additional context

_No response_

### LINKED ISSUE #8708: reth-primitives no-std support

### PR DESCRIPTION
**Pull request by JackG-eth**
Motivation
closes #8708
closes #5746

### CODE REVIEW

**Comment by JackG-eth** [comment]
Might be a couple of Nits here, a lot more has changed @mattsse

File: crates/primitives/src/account.rs [hunk 1]
```diff
@@ -4,8 +4,12 @@ use bytes::Buf;
 use reth_codecs::Compact;
 use revm_primitives::JumpTable;
 use serde::{Deserialize, Serialize};
+#[cfg(feature = "std")]
 use std::ops::Deref;
 
+#[cfg(not(feature = "std"))]
+use core::ops::Deref;
```
Comment (DaniPopes) [inline crates/primitives/src/account.rs:11]: Please just use `core` as it's always available.

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/primitives/src/account.rs (hunk 1 above)
Comment (JackG-eth) [reply crates/primitives/src/account.rs:11]: for any case where we have both std/core import for the same thing?

Comment (DaniPopes) [reply crates/primitives/src/account.rs:11]: yes, std just re-exports core internally for most items

**Review by mattsse — CHANGES_REQUESTED** [review]
we just extracted a lot of stuff, sorry about the conflicts -.-

**Comment by JackG-eth** [comment]
This Is WIP after rebasing..

**Comment by JackG-eth** [comment]
This should be good now, both builds work with conflicts resolved.

**Review by mattsse — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**